### PR TITLE
Error handling for fragment processing + FragmentHandler integration …

### PR DESCRIPTION
**Pull request by marcinkp**
…test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the [code style](/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the Knot.x Contributor License Agreement.

### CODE REVIEW

File: core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+@ExtendWith(VertxExtension.class)
+class FragmentsHandlerTest {
+
+  @Test
+  @DisplayName("Expect Status Code 200 when task completes on _success.")
+  public void shouldSuccess(Vertx vertx, VertxTestContext testContext)
+      throws IOException {
+    //given
+    int port = 9998;
```
Comment (Skejven) [inline core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java]: Please use here [random ports mechanics](https://github.com/Knotx/knotx-junit5#randomport).

File: core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    JsonObject config = from("tasks/fragment-handler/successAction.json");
+    FragmentsHandler underTest = new FragmentsHandler(vertx, config);
+    mockServer(vertx, underTest, fragment("success-task"), port);
+
+    //when
+    WebClient.create(vertx)
```
Comment (Skejven) [inline core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java]: Wouldn't be much easier to make `toHandlerResult` package protected and unit test it here without creating a server?

Such check with the whole server is a very good case for the integration tests in the `knotx-stack`.

Comment (tomaszmichalak) [reply core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java]: I think we should add contract tests that validate the output of handler based on the input.

File: core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+import io.vertx.reactivex.ext.web.RoutingContext;
+import io.vertx.reactivex.ext.web.client.WebClient;
+
+@ExtendWith(VertxExtension.class)
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class FragmentsHandlerTest {
+
+  @Test
+  void shouldFail(Vertx vertx, VertxTestContext testContext)
```
Comment (tomaszmichalak) [inline core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java]: PLease add display name.

File: core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+import io.vertx.reactivex.ext.web.RoutingContext;
+import io.vertx.reactivex.ext.web.client.WebClient;
+
+@ExtendWith(VertxExtension.class)
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class FragmentsHandlerTest {
+
+  @Test
+  @DisplayName("Expect fail with Status Code 500 when task completes on non exist transition.")
```
Comment (tomaszmichalak) [inline core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java]:
```suggestion
  @DisplayName("Expect fail with Status Code 500 when task completes with fragment that is failed.")
```

File: core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    assertTrue(testContext.awaitCompletion(5, TimeUnit.SECONDS));
+
+    if (testContext.failed()) {
+      throw testContext.causeOfFailure();
+    }
+
+  }
+
+  @Test
+  @DisplayName("Expect continuing processing next handler when task completes on _success.")
```
Comment (tomaszmichalak) [inline core/src/test/java/io/knotx/fragments/handler/FragmentsHandlerTest.java]:
```suggestion
  @DisplayName("Expect continuing processing next handler when task completes with all fragments in status success or unprocessed")
```

**Pull request merged**